v2/dtos/deviceresource.go

```diff
@@ -0,0 +1,30 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package dtos
+
+import "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+// DeviceResource represents a value on a device that can be read or written
+// This object and its properties correspond to the DeviceResource object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/DeviceResource
+type DeviceResource struct {
+	Description string            `json:"description" yaml:"description,omitempty"`
+	Name        string            `json:"name" yaml:"name,omitempty"`
```

Comment:
Name and other  fields?? should be required since it is used in a Add request

Reply:
Added name as a required field.